index.js

```diff
@@ -60,12 +60,13 @@ minimum value for how long a non-cached analyses will take
     *                              this must be larger than defaultInitialDelay which we believe to be
     *                              the smallest reasonable value.
     *
+    * @param {boolean} uuid        - optional, when set true includes uuid of analysis in results.
```

Comment:
Here is a summary of the discussion that @daniyarchambylov and I had over this. 

The main thing is that `analysis()`, while it was once the external interface, is now deprecated and should be internal only. Adding the UUID into an object (that doesn't have `typeof(object) === 'Array'`, but seems to function like one) was simply a workaround for what feels to me the larger misfeature that the API should never have returned just a list of issues but should have always included the UUID so that you could reference that with other information the API returns. 

In the long term I would like to see API changed to include the UUID when issues are returned, possibly by adding a new endpoint and depricating this one. Independent of that, in this library, `analysis()` would get replaced with `analysisWithStatus()` with some brief period of time that the two are identical for compatibilty.

For the short term though, adding an optional parameter complicates things. If someone wants to use analysis() externally (and they should not), then they'll have to live with getting the "ugliness" of a UUID in an array-like object that should have not been simply an array-like object.

I explored the possibility of removing the export of `analysis()`. However Daniyar says that would cause breakage and better to open an issue to fix all of this properly, pusing out in another 
major release in the future, but not this one. 

@daniyarchambylov would you open a ticket so we don't forget to ultimately fix this correctly?

Reply:
https://github.com/ConsenSys/armlet/issues/70